### ISSUE TITLE
SNS topic can't be created for bucket with period in name

### ISSUE THREAD

**Issue by vt102** [kislyuk/domovoi#5]
I have asset buckets related to static sites, with the domain name in the bucket name; e.g., www.example.com-uploads.

Domovai fails when creating an SNS topic with the bucket name in the SNS topic name, as SNS topic are not allows to have periods in the name; e.g., trying to create an SNS topic with name domovoi-s3-bucket-events-www.example.com-upload fails, `Topic name contains invalid characters. Must contain only alphanumeric characters, hyphens (-), or underscores (_).`

For example:

```
@app.s3_event_handler(bucket='www.example.com-upload', events=['s3:ObjectCreated:*'])
def resize_image(event, context):
     return True
```

```
andy@shoggoth:dosizer$ domovoi deploy
Regen deployment package.
:
Found existing permission grant statement domovoi-17ccb624, skipping
Subscribing {'prefix': None, 'suffix': None, 'events': ['s3:ObjectCreated:*'], 'func': <function resize_image at 0x7efc4c523aa0>, 'use_sns': True} to events in S3 bucket www.example.com-upload
Traceback (most recent call last):
  File "/home/andy/.local/bin/domovoi", line 168, in <module>
    sns_topic = sns.create_topic(Name="domovoi-s3-bucket-events-{}".format(s3_bucket))
  File "/home/andy/.local/lib/python2.7/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/home/andy/.local/lib/python2.7/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/home/andy/.local/lib/python2.7/site-packages/botocore/client.py", line 317, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/andy/.local/lib/python2.7/site-packages/botocore/client.py", line 615, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.InvalidParameterException: An error occurred (InvalidParameter) when calling the CreateTopic operation: Invalid parameter: Topic Name
```

**Comment by kislyuk** [comment]
Thanks for reporting! Will fix this shortly. Just need to pick a way to
escape the periods.

**Comment by kislyuk** [comment]
S3 bucket names cannot contain underscores, while SNS topic names cannot contain periods. So I'll use that replacement as a reversible name transform.

**Comment by kislyuk** [comment]
Released in v1.5.2.
